### ISSUE TITLE
getEnv() must be an instance of PhilKra\Events\object error

### ISSUE THREAD

**Issue by shabananavas** [philkra/elastic-apm-php-agent#54]
I'm getting the following error when 'env' is an empty array.

`NOTICE: PHP message: TypeError: Return value of PhilKra\Events\EventBean::getEnv() must be an instance of PhilKra\Events\object, instance of stdClass returned in /app/vendor/philkra/elastic-apm-php-agent/src/Events/EventBean.php on line 183`

The same error happens for cookies as well.

**Comment by philkra** [comment]
oh! thanks for reporting this! can you please post your config array?

**Comment by shabananavas** [comment]
My config array looks like this:

<img width="825" alt="screen shot 2019-02-26 at 10 17 25 pm" src="https://user-images.githubusercontent.com/11439056/53430408-70ef5200-3a14-11e9-8cc7-4bf50e9c0d6e.png">

**Comment by dstepe** [comment]
I'm not sure this is an issue in the package. That seems like the cast to an object is not working, which is a language problem. The error indicates it's looking for a class named 'object' in the current class's namespace. What version of PHP are you using?

Can you try a similar cast in otherwise working code? Something like:

```
var_dump((object) []);
```
You should get something like:

```
object(stdClass)#250 (0) { }
```

**Comment by shabananavas** [comment]
I'm running PHP 7.1.26 and this is what I see when I cast the array as an object:

`class stdClass#251 (0) { }`

**Comment by dstepe** [comment]
Wow. I completely missed the problem last night. It's the return type of that method declaration. The `object` type hint was introduced in PHP 7.2:

http://php.net/manual/en/migration72.new-features.php

So yes, this would be a problem under 7.1 and earlier.

@philkra What is your position on minimum supported PHP versions? The composer file asserts `>= 7.0`.

**Comment by philkra** [comment]
@dstepe thanks for the investigation and identification! I would really like to keep the `>= 7.0` requirement. I'll fix this to work for `7.0`

**Comment by dstepe** [comment]
I just finished building a compatibility check into our pipeline using phpcs. I ran it on the `src` directory testing for 7.0. (Ignore the `/src/src`. That's a result of how the code is mounted into the Docker container.)

```
Scanning src for PHP 7.0 compatibility
........W...E........ 21 / 21 (100%)



FILE: /src/src/Events/Transaction.php
-------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
-------------------------------------------------------------------------------------------------------------------------------------------------------
 86 | WARNING | Since PHP 7.0, functions inspecting arguments, like debug_backtrace(), no longer report the original value as passed to a parameter,
    |         | but will instead provide the current value. The parameter "$duration" was used, and possibly changed (by reference), on line 84.
-------------------------------------------------------------------------------------------------------------------------------------------------------


FILE: /src/src/Events/EventBean.php
-------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
-------------------------------------------------------------------------------
 176 | ERROR | object return type is not present in PHP version 7.1 or earlier
 193 | ERROR | object return type is not present in PHP version 7.1 or earlier
-------------------------------------------------------------------------------

Time: 387ms; Memory: 8MB
```

**Comment by philkra** [comment]
I pushe da fix to Development branch

**Comment by philkra** [comment]
This is fixed with the Agents Release ˋv6.5.4ˋ
